src/main/java/com/linkedin/kmf/services/ClusterTopicManipulationService.java

```diff
@@ -0,0 +1,248 @@
+
+    private ClusterTopicManipulationServiceRunnable() {
+      // unaccessed.
+    }
+
+    /**
+     * When an object implementing interface <code>Runnable</code> is used
+     * to create a thread, starting the thread causes the object's
+     * <code>run</code> method to be called in that separately executing
+     * thread.
+     * <p>
+     * The general contract of the method <code>run</code> is that it may
+     * take any action whatsoever.
+     *
+     * @see     Thread#run()
+     */
+    @Override
+    public void run() {
+      try {
+        ClusterTopicManipulationService.this.createDeleteClusterTopic();
+      } catch (Exception e) {
+        LOGGER.error("{} {} failed to run createDeleteClusterTopic()", _configDefinedServiceName,
+            ClusterTopicManipulationService.this.getClass().getSimpleName(), e);
+      }
+    }
+  }
+
+  private void createDeleteClusterTopic() {
+
+    if (_isOngoingTopicCreationDone) {
+
+      int random = ThreadLocalRandom.current().nextInt();
+      _currentlyOngoingTopic = XinfraMonitorConstants.TOPIC_MANIPULATION_SERVICE_TOPIC + Math.abs(random);
+
+      try {
+        int brokerCount = _adminClient.describeCluster().nodes().get().size();
+        _createTopicsResult = _adminClient.createTopics(Collections.singleton(
+            new NewTopic(_currentlyOngoingTopic, XinfraMonitorConstants.TOPIC_MANIPULATION_TOPIC_NUM_PARTITIONS,
+                (short) brokerCount)));
+
+        _totalPartitions.set(brokerCount * XinfraMonitorConstants.TOPIC_MANIPULATION_TOPIC_NUM_PARTITIONS);
+
+        _isOngoingTopicCreationDone = false;
+        LOGGER.trace("Initiated a new topic creation. topic information - topic: {}, cluster broker count: {}",
+            _currentlyOngoingTopic, brokerCount);
+      } catch (InterruptedException | ExecutionException e) {
+        LOGGER.error("Exception occurred while retrieving the brokers count: ", e);
+      }
+    }
+
+    try {
+      LOGGER.trace("cluster id: {}", _adminClient.describeCluster().clusterId().get());
+      Collection<Node> brokers = _adminClient.describeCluster().nodes().get();
+      if (_createTopicsResult.all().isDone() && doesClusterContainTopic(_currentlyOngoingTopic, brokers,
+          _adminClient)) {
+        _adminClient.deleteTopics(Collections.singleton(_currentlyOngoingTopic)).all();
+        LOGGER.trace("clusterTopicManipulationServiceRunnable: Initiated topic deletion on {}.",
+            _currentlyOngoingTopic);
+        _isOngoingTopicCreationDone = true;
+        LOGGER.trace("{}-clusterTopicManipulationServiceRunnable successful!", this.getClass().getSimpleName());
+      }
+    } catch (InterruptedException | ExecutionException e) {
+      LOGGER.error("Exception occurred while creating cluster topic in {}: ", _configDefinedServiceName, e);
+    }
+  }
+
+  private boolean doesClusterContainTopic(String topic, Collection<Node> brokers, AdminClient adminClient)
```

Comment:
Need Java doc to explain how it checks whether a cluster contains a topic or not. It is not very intuitive to understand.

Question: shouldn't we check whether a cluster contains a topic by fetching metadata instead of checking brokers' log directories? Because a topic does not exist until its metadata is in the results of the fetch metadata request. If KMF is measuring how long it takes to create and delete a topic from the user's PoV, it should look at the fetched metadata instead of the log dirs on Kafka brokers;

Reply:
It's true that metadata fetch is useful. Does AdminClient provide fetching metadata of brokers? 

If that was the case, rather than `instead` of checking brokers' log dirs, it should be fetching metadata `and` checking brokers' log directories. We should add metadata fetch if possible in the next PR.

Reply:
I quickly checked the `AdminClient` API. I don't think there is a method that lets us send a fetch metadata request to the Kafka server. Here are 2 things you can consider.

1. Use the `describeTopics` methods. The idea is to keep calling `describeTopics` to describe the topic you are trying to create until you see this method returns results for the created topic. Same thing for deleted topics. Keep describing deleted topics until you see it stops returning information about the deleted topics.

2. Construct a fetch metadata request yourself and send it to Kafka broker. I am not sure how hard it would be. But worth exploring.

Overall IMO, **checking whether brokers' log dirs indicate topic creation or topic deletion is almost meaningless (sorry for making it sound harsh) since that's NOT what users see.** We all know that there is some delay after log dirs created for a topic until users see the topic in metadata and then be able to use the topic. This delay might vary and definitely worth measuring. You can choose to implement fetching metadata in the next PR. But I do NOT agree that we should keep this approach and use it in production since it is meaningless.

What do you think @smccauliff ?

Reply:
Got back from lunch. I believe the remaining topic to be discussed is :

```I quickly checked the AdminClient API. I don't think there is a method that lets us send a fetch metadata request to the Kafka server. Here are 2 things you can consider.
Use the describeTopics methods. The idea is to keep calling describeTopics to describe the topic you are trying to create until you see this method returns results for the created topic. Same thing for deleted topics. Keep describing deleted topics until you see it stops returning information about the deleted topics.
Construct a fetch metadata request yourself and send it to Kafka broker. I am not sure how hard it would be. But worth exploring.```

As I commented right before your reply, `If that was the case, rather than instead of checking brokers' log dirs, it should be fetching metadata and checking brokers' log directories. We should add metadata fetch if possible in the next PR.` , could we include the the fetch metadata or the AdminClient describeTopics as part of the next PR?

I will not ELR this commit with the next release. I am happy to adopt one of the two approaches you suggested.